capstone-rs/src/instruction.rs

```diff
@@ -36,6 +37,64 @@ pub type RegIdInt = u16;
 #[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct RegId(pub RegIdInt);
 
+/// Represents how the register is accessed.
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub enum RegAccessType {
+    /// Operand read from memory or register.
+    Read,
+    /// Operand write from memory or register.
+    Write,
+    /// Operand read and write from memory or register.
+    ReadWrite,
+}
+
+impl RegAccessType {
+    /// Returns whether the instruction reads from the operand.
+    ///
+    /// Note that an instruction may read and write to the register
+    /// simultaneously. In this case, the operand is also considered as
+    /// readable.
+    pub fn is_readable(&self) -> bool {
+        if let RegAccessType::Read | RegAccessType::ReadWrite = self {
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Returns whether the instruction writes from the operand.
+    ///
+    /// Note that an instruction may read and write to the register
+    /// simultaneously. In this case, the operand is also considered as
+    /// writable.
+    pub fn is_writable(&self) -> bool {
+        if let RegAccessType::Write | RegAccessType::ReadWrite = self {
+            true
+        } else {
+            false
+        }
+    }
+}
+
+impl TryFrom<cs_ac_type> for RegAccessType {
+    type Error = ();
+
+    fn try_from(access: cs_ac_type) -> Result<Self, Self::Error> {
+        if access == CS_AC_INVALID {
+            return Err(());
+        }
```

Comment:
IMHO, checking it first matches the semantics better, because the API of capstone only specifies that `CS_AC_INVALID` means "invalid". It's possible that an access is neither readable nor writable, semantically. (Although this scenario is not possible with current `capstone.h`)